backend/src/modules/match-module/core/domain/Match.ts

```diff
@@ -0,0 +1,63 @@
+import { MatchHasStarted } from './event/MatchHasStarted';
+import { MatchId } from './MatchId';
+import { MatchSideId } from './MatchSideId';
+import { DomainCommandResult } from '../../../../shared/core/domain/DomainCommandResult';
+
+export class Match {
+  readonly matchId: MatchId;
+  readonly firstTeamId: MatchSideId;
+  readonly secondTeamId: MatchSideId;
+  readonly winner: MatchSideId | undefined;
+  readonly looser: MatchSideId | undefined;
+  readonly hasEnded: boolean | undefined = false;
+
+  constructor(props: {
+    matchId: MatchId;
+    firstTeamId: MatchSideId;
+    secondTeamId: MatchSideId;
+    winner?: MatchSideId;
+    looser?: MatchSideId;
+    hasEnded?: boolean;
+  }) {
+    this.matchId = props.matchId;
+    this.firstTeamId = props.firstTeamId;
+    this.secondTeamId = props.secondTeamId;
+    this.winner = props.winner;
+    this.looser = props.looser;
+    this.hasEnded = props.hasEnded;
+  }
+}
+
+export function startMatch(
+  state: Match | undefined,
+  command: { matchId: string; firstTeamId: string; secondTeamId: string },
```

Comment:
Tutaj dobrze już jakby wchodziły typy domenowe. Konwersja z zewnątrz na zrozumiale dla domeny zajmuje się właśnie aplikacja :) Ta operacja biznesowa już powinna założyć, że ma wszystko poprawne. Bo teraz jakbyś zmienił coś w command (application) leci to aż tutaj. Zobacz sobie na tournament registartions :)

Reply:
spoko, zaraz nareperuję :D